source/server-discovery-and-monitoring/tests/unified/backpressure-network-error-fail.yml

```diff
@@ -0,0 +1,97 @@
+description: backpressure-network-error-fail
+schemaVersion: "1.17"
+runOnRequirements:
+  - minServerVersion: "4.4"
+    serverless: forbid
+    topologies:
+      - single
+      - replicaset
+      - sharded
+createEntities:
+  - client:
+      id: setupClient
+      useMultipleMongoses: false
+initialData:
+  - collectionName: backpressure-network-error-fail
+    databaseName: sdam-tests
+    documents:
+      - _id: 1
+      - _id: 2
+tests:
+  - description: apply backpressure on network connection errors during connection establishment
+    operations:
+      - name: createEntities
+        object: testRunner
+        arguments:
+          entities:
+            - client:
+                id: client
+                useMultipleMongoses: false
+                observeEvents:
+                  - serverHeartbeatSucceededEvent
+                  - serverDescriptionChangedEvent
+                  - poolReadyEvent
+                  - poolClearedEvent
+                uriOptions:
+                  retryWrites: false
+                  heartbeatFrequencyMS: 10000
+                  serverMonitoringMode: poll
+                  directConnection: true
+                  appname: backpressureNetworkErrorFailTest
+            - database:
+                id: database
+                client: client
+                databaseName: sdam-tests
+            - collection:
+                id: collection
+                database: database
+                collectionName: backpressure-network-error-fail
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: client
+          event:
+            serverHeartbeatSucceededEvent: {}
+          count: 1
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          client: client
+          event:
+            serverDescriptionChangedEvent: {}
+          count: 1
```

Comment:
out of curiousity - what's the reason we need to wait for a `serverHeartbeatSucceededEvent` before we wait for a `serverDescriptionChangedEvent`?  Wouldn't waiting on only an `serverDescriptionChangedEvent` work?